synapse/handlers/user_directory.py

```diff
@@ -464,6 +464,10 @@ async def _handle_possible_remote_profile_change(
 
         prev_name = prev_event.content.get("displayname")
         new_name = event.content.get("displayname")
+
+        # Replace any NULL characters in the name as these cannot be stored in the database
+        new_name = new_name.replace("\x00", "\uFFFD")
```

Comment:
For completeness, sqlite sort-of-supports null-codepointsin strings, with scary caveats:  https://sqlite.org/nulinstr.html
Postgres definitely doesn't and won't any time soon. See e.g. [this HN post](https://news.ycombinator.com/item?id=25976266).